### DISCUSSION TITLE
How to get the current page URL including arguments?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#769] [Ask a question] [answered]
Hello,

I need help with getting the current URL. Lets say we have `https://tld.de/tag:example/`. When I use `<?php echo $this->yellow->page->getUrl() ?>` in a layout file I get `https://tld.de/` without `tag:example`. I want to get the full URL with arguments. Is there a solution with the Yellow API?

Best regards

**Comment by PetersOtto** [comment]
Oh, I think I found a solution myself.

`<?php echo 'tag:' . preg_replace('/^.*: /', '', strtolower($this->yellow->page->getHtml("TitleContent"))) ?>`

Is it a good solution or would you prefer a other way?

Update 31.03.2022:
--------------------------------
Me helps this as well: https://stackoverflow.com/questions/6768793/get-the-full-url-in-php

**Comment by markseuffert** [comment, marked as answer]
You could use the following for the URL including arguments:

```
<?php echo $this->yellow->page->getUrl().$this->yellow->toolbox->getLocationArguments() ?>
```

Datenstrom Yellow uses [clean URLs](https://en.wikipedia.org/wiki/Clean_URL). Instead of passing arguments with query strings it uses location arguments. Instead of `https://example.com/search?query=yellow` it  uses `https://example.com/search/query:yellow/`. Perhaps the API documentation needs a few more examples for developers.

Hope this helps.
